src/coreComponents/common/StdContainerWrappers.hpp

```diff
@@ -66,83 +92,129 @@ class StdVectorWrapper : public std::vector< T, Allocator >
-//           typename T,
-//           typename Hash = std::hash<Key>,
-//           typename KeyEqual = std::equal_to<Key>,
-//           typename Allocator = std::allocator<std::pair<const Key, T>>,
-//           bool USE_STD_CONTAINER_BOUNDS_CHECKING = false>
-// using unordered_map = internal::StdMapWrapper< std::unordered_map< Key, T, Hash, KeyEqual, Allocator >,
-//                                                USE_STD_CONTAINER_BOUNDS_CHECKING >;
+  MappedType const & operator[]( KeyType const & key ) const
+  {
+    if constexpr (USE_BOUNDS_CHECKING)
+    {
+      return this->at( key );
+    }
+    else
+    {
+      return Base::operator[]( key );
+    }
+  }
+};
 
+} //namespace internal
+
+/**
+ * type alias for std::map
+ * @tparam Key The unique std::map key.
+ * @tparam T Type of elements in the std::map.
+ * @tparam Compare The comparison function used to order the keys. Defaults to std::less<Key>.
+ * @tparam Allocator Allocator type for the map. Defaults to std::allocator<std::pair<const Key, T>>
+ */
+template< typename Key,
+          typename T,
+          typename Compare = std::less< Key >,
+          typename Allocator = std::allocator< std::pair< const Key, T > > >
+using stdMap = internal::StdMapWrapper< std::map< Key, T, Compare, Allocator >,
+                                        USE_STD_CONTAINER_BOUNDS_CHECKING >;
+
+/**
+ * type alias for std::unordered_map
+ * @tparam Key The unique std::unordered_map key.
+ * @tparam T Type of elements in the std::unordered_map.
+ * @tparam Hash The hash function to be used for the keys. Defaults to std::hash<Key>
+ * @tparam KeyEqual The function used to compare keys for equality. Defaults to std::equal_to<Key>.
+ * @tparam Allocator Allocator type for the map.  Defaults to std::allocator<std::pair<const Key, T>>
+ */
+template< typename Key,
+          typename T,
+          typename Hash = std::hash< Key >,
+          typename KeyEqual = std::equal_to< Key >,
+          typename Allocator = std::allocator< std::pair< const Key, T > > >
+using stdUnorderedMap = internal::StdMapWrapper< std::unordered_map< Key, T, Hash, KeyEqual, Allocator >,
+                                                 USE_STD_CONTAINER_BOUNDS_CHECKING >;
+
+/**
+ * @name Ordered and unordered map types.
+ */
+///@{
+
+/**
+ * @brief Base template for ordered and unordered maps.
+ * @tparam TKEY key type
+ * @tparam TVAL value type
+ * @tparam SORTED a bool indicating whether map is ordered
+ */
+template< typename TKEY, typename TVAL, typename SORTED >
+class mapBase
+{};
+
+/// @cond DO_NOT_DOCUMENT
+template< typename TKEY, typename TVAL >
+class mapBase< TKEY, TVAL, std::integral_constant< bool, true > > : public stdMap< TKEY, TVAL >
+{
+public:
+  using stdMap< TKEY, TVAL >::stdMap;
+};
+
+template< typename TKEY, typename TVAL >
+class mapBase< TKEY, TVAL, std::integral_constant< bool, false > > : public stdUnorderedMap< TKEY, TVAL >
+{
+  using stdUnorderedMap< TKEY, TVAL >::stdUnorderedMap;
+};
```

Comment:
- Do you think that would possible to rename mapBase to `mapType` (it isn't really the base class of anything)?
- Couldn't the `SORTED` parameter be a simple `bool`?
- I would use those types only when strictly needed.
```suggestion
template< typename TKEY, typename TVAL, bool SORTED >
class mapType
{};

/// @cond DO_NOT_DOCUMENT
template< typename TKEY, typename TVAL >
class mapType< TKEY, TVAL, true > : public stdMap< TKEY, TVAL >
{
public:
  using stdMap< TKEY, TVAL >::stdMap;
};

template< typename TKEY, typename TVAL >
class mapType< TKEY, TVAL, false > : public stdUnorderedMap< TKEY, TVAL >
{
  using stdUnorderedMap< TKEY, TVAL >::stdUnorderedMap;
};
```